tests/test_cli.py

```diff
@@ -0,0 +1,220 @@
+#!/usr/bin/env python3
+
+from pytest import fixture
+from subprocess import Popen, PIPE
+from os import getcwd, X_OK, access
+from pathlib import Path
+
+
+class CLITestDefaultValues:
+    GRINDER_RUN_COMMAND = "./grinder.py"
+    CLI_QUERY_PATH = "tests/test_data/test_queries/cli_arguments_test.json"
+    if Path(getcwd()).stem == "tests":
+        GRINDER_RUN_COMMAND = f".{GRINDER_RUN_COMMAND}"
+        CLI_QUERY_PATH = CLI_QUERY_PATH[len(Path(getcwd()).stem) + 1 :]
+    if not access("grinder.py", X_OK) and not access("../grinder.py", X_OK):
+        if Path(getcwd()).stem == "tests":
+            GRINDER_RUN_COMMAND = f"python3 {GRINDER_RUN_COMMAND}"
+        else:
+            GRINDER_RUN_COMMAND = f"python3 {GRINDER_RUN_COMMAND[2:]}"
```

Comment:
Okay, thanks, I see that you fix it.
Please, check that it is works as expected. I think this solution can be more readable.
```python3
#!/usr/bin/env python3

from os import path, access, X_OK
from pathlib import Path


def check_executable_flag(file_to_check: str = "grinder.py") -> str:
    """
    Find the original grinder.py and check if it executable
    :param file_to_check: filename of module, "grinder.py" by default
    :return: string representation of command to run grinder
    """
    try:
        # resolve full path to test + get parent of parent (e.g. root directory of grinder)
        grinder_dir = Path(__file__).resolve().parents[1]
    except IndexError:
        # if inner structure of directories is corrupted
        raise Exception(
            "Wrong path. Please, check that tests are located in grinder root directory."
        )
    full_grinder_path = grinder_dir.joinpath(file_to_check)
    if not access(full_grinder_path, X_OK):
        return f"python3 {full_grinder_path}"
    return f"{full_grinder_path}"


class CLITestDefaultValues:
    CLI_QUERY_PATH = "tests/test_data/test_queries/cli_arguments_test.json"
    GRINDER_RUN_COMMAND = check_executable_flag()
    print(GRINDER_RUN_COMMAND)

```

Results:
1. Run it from `tests` directory with chmod +x:
```
pytest test_exec_grinder.py -s -v
collecting ... /Users/a.nikolaev/Grinder/grinder/grinder.py
```
2. Run it from `grinder` directory with chmod +x:
```
pytest tests/test_exec_grinder.py -s -v
collecting ... /Users/a.nikolaev/Grinder/grinder/grinder.py
```
3. Run if from the parent directory, upper then grinder root directory itself:
```
pytest grinder/tests/test_exec_grinder.py -s -v
collecting ... /Users/a.nikolaev/Grinder/grinder/grinder.py
```
4. Run it from `tests` directory without chmod +x:
```
pytest test_exec_grinder.py -s -v
collecting ... python3 /Users/a.nikolaev/Grinder/grinder/grinder.py
```

Reply:
Also i think that `CLI_QUERY_PATH` will be always fixed in this case - please, check it.